internal/resources/sample_bucket.go

```diff
@@ -0,0 +1,351 @@
+	}
+}
+
+func (r *SampleBucket) validateCreateBucket(plan providerschema.Bucket) error {
+	if plan.OrganizationId.IsNull() {
+		return errors.ErrOrganizationIdMissing
+	}
+	if plan.ProjectId.IsNull() {
+		return errors.ErrProjectIdMissing
+	}
+	if plan.ClusterId.IsNull() {
+		return errors.ErrClusterIdMissing
+	}
+	return r.validateBucketName(plan)
+}
+
+func (r *SampleBucket) validateBucketName(plan providerschema.Bucket) error {
+	if (!plan.Name.IsNull() && !plan.Name.IsUnknown()) && !providerschema.IsTrimmed(plan.Name.ValueString()) {
+		return fmt.Errorf("name %s", errors.ErrNotTrimmed)
+	}
+
+	if !isValidSampleName(plan.Name.ValueString()) {
+		return errors.ErrInvalidSampleBucketName
+	}
+
+	return nil
+}
+
+// retrieveBucket retrieves bucket information for a specified organization, project, cluster and bucket ID.
+func (s *SampleBucket) retrieveBucket(ctx context.Context, organizationId, projectId, clusterId, bucketId string) (*providerschema.OneBucket, error) {
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/sampleBuckets/%s", s.HostURL, organizationId, projectId, clusterId, bucketId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := s.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		s.Token,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrExecutingRequest, err)
+	}
+
+	bucketResp := samplebucket.GetSampleBucketResponse{}
+	err = json.Unmarshal(response.Body, &bucketResp)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrUnmarshallingResponse, err)
+	}
+	refreshedState := providerschema.OneBucket{
+		Id:                       types.StringValue(bucketResp.Id),
+		Name:                     types.StringValue(bucketResp.Name),
+		OrganizationId:           types.StringValue(organizationId),
+		ProjectId:                types.StringValue(projectId),
+		ClusterId:                types.StringValue(clusterId),
+		Type:                     types.StringValue(bucketResp.Type),
+		StorageBackend:           types.StringValue(bucketResp.StorageBackend),
+		MemoryAllocationInMB:     types.Int64Value(bucketResp.MemoryAllocationInMb),
+		BucketConflictResolution: types.StringValue(bucketResp.BucketConflictResolution),
+		DurabilityLevel:          types.StringValue(bucketResp.DurabilityLevel),
+		Replicas:                 types.Int64Value(bucketResp.Replicas),
+		Flush:                    types.BoolValue(bucketResp.Flush),
+		TimeToLiveInSeconds:      types.Int64Value(bucketResp.TimeToLiveInSeconds),
+		EvictionPolicy:           types.StringValue(bucketResp.EvictionPolicy),
+		Stats: &providerschema.Stats{
+			ItemCount:       types.Int64Value(bucketResp.Stats.ItemCount),
+			OpsPerSecond:    types.Int64Value(bucketResp.Stats.OpsPerSecond),
+			DiskUsedInMiB:   types.Int64Value(bucketResp.Stats.DiskUsedInMib),
+			MemoryUsedInMiB: types.Int64Value(bucketResp.Stats.MemoryUsedInMib),
+		},
+	}
+
+	return &refreshedState, nil
+}
+
```

Comment:
Ideally, we should not use two different structs for request and response. For instance, using providerschema.Bucket for the request and providerschema.OneBucket in the response. Instead, both should consistently use the same struct, such as providerschema.Bucket.

Reply:
Why have we done that for buckets then? I was following that as an example as the get response for sample buckets and buckets is identical.

Reply:
Indeed, the Bucket implementation appears to have drawn inspiration from the project implementation. When the project implementation was initially written, there were some issues associated with using the same struct. However, these issues were subsequently addressed in the later resources. Unfortunately, during the bucket resource creation, this improvement seems to have been missed in the review process.

Reply:
will create a new ticket for improvements

Reply:
I've already raised a ticket for it https://couchbasecloud.atlassian.net/browse/AV-74460